docs/concepts/projections-extents-layouts.md

```diff
@@ -0,0 +1,162 @@
+referencing the Geotrellis objects and source files backing them.
+
+This document aims to be informative to new and experienced GIS users alike.
+
+Projections
+-----------
+
+**Data structures:** `CRS`, `LatLng`, `WebMercator`, `ConusAlbers`
+
+**Sources:** `geotrellis.proj4.{CRS, LatLng, WebMercator, ConusAlbers}`
+
+**What is a projection?**
+
+In GIS, a *projection* is a mathematical transformation of
+Latitude/Longitude coordinates on a sphere onto some other flat plane. Such
+a plane is naturally useful for representing a map of the earth in 2D. A
+projection is defined by a *Coordinate Reference System* (CRS), which holds
+some extra information useful for reprojection. CRSs themselves have static
+definitions, have agreed-upon string representations, and are usually made
+public by standards bodies or companies. They can be looked up at
+[SpatialReference.org](http://spatialreference.org/).
+
+A *reprojection* is the transformation of coorindates in one CRS to another.
+To do so, coordinates are first converted to those of a sphere. Every CRS
+knows how to convert between its coordinates and a sphere's, so a
+transformation `CRS.A -> CRS.B -> CRS.A` is actually `CRS.A -> Sphere ->
+CRS.B -> Sphere -> CRS.A`. Naturally some floating point error does
+accumulate during this process.
+
+Within the context of Geotrellis, the main projection-related object is the
+`CRS` trait. It stores related `CRS` objects from underlying libraries, and
+also provides the means for defining custom reprojection methods, should the
+need arise. It's companion object provides convenience functions for
+creating `CRS`s. Geotrellis currently has three `object`s that implement the
+`CRS` trait: `LatLng`, `WebMercator`, and `ConusAlbers`.
+
+**What can CRSs do?**
+
+They can be transformed back into their String representations:
+
+self => toWKT, toProj4String
+
+**How are CRSs used throughout Geotrellis?**
+
+`CRS`s are stored in the `*ProjecedExtent` classes and are used chiefly
+to define how reprojections should operate. Example:
+
+```scala
+val wm = Line(...)  // A `LineString` vector object in WebMercator.
+val ll: Line = wm.reproject(WebMercator, LatLng)  // The Line reprojected into LatLng.
+
+```
+
+Extents
+-------
+
+**Data structures:** `Extent`, `ProjectedExtent`, `TemporalProjectedExtent`,
+`GridExtent`, `RasterExtent`
+
+**Sources:** `geotrellis.vector.Extent`,
+`geotrellis.vector.reproject.Reproject`,
+`geotrellis.spark.TemporalProjectExtent`,
+`geotrellis.raster.{ GridExtent, RasterExtent }`,
+`geotrellis.raster.reproject.ReprojectRasterExtent`
+
+**What is an extent?**
+
+An `Extent` is a rectangular section of a 2D projection of the Earth. It is
+represented by two coordinate pairs that are its "min" and "max" corners in
+some Coorindate Reference System. "min" and "max" here are CRS
+specific, as the location of the point `(0,0)` varies between different CRS.
+An Extent can also be referred to as a *Bounding Box*.
```

Comment:
The whole extent is CRS specific; anything relating to the Extent will have to be in that CRS, and users of the extent should know the CRS of the extent if they are trying to combine it with other things, either by knowing implicitly that all the geometries they are working with are in the same CRS, or explicitly by holding onto CRS instances, and reprojecting where necessary.